File: cypress/cypress/integration/Username.js
import loginPage from '../support/pages/loginPage';
import creationPage from "../support/pages/creationPage";
import defaultValues from "../support/shared/defaultValues"

describe('Username', () => {
    beforeEach(() => {
        loginPage.FillAndSubmit();
        creationPage.goToCreationPage();
    })
  
    it('RF_01 - Caracteres permitidos', () => {
       creationPage.inputUsername(true);
       cy.get("[data-valmsg-for=Username]").should('be.empty') ;
       creationPage.inputUsername(false);
       cy.get("[data-valmsg-for=Username]").contains(defaultValues.InvalidUsername);
    })

    it('RF_02 - Prefijos no permitidos', () => {
      
        creationPage.inputSpecificUsername("endpoint");
        cy.get("[data-valmsg-for=Username]").contains(defaultValues.InvalidUsername);
        
        cy.get("#Username").clear();
        creationPage.inputSpecificUsername("anyone");
        //cy.wait(1000);
       cy.get(".field-validation-error > span").contains(defaultValues.InvalidUsername);
    })

    it('RF_04 - Longitud permitida', () => {
        cy.get("#Username").clear();
        creationPage.inputUsername(true,65);
        cy.wait(1000);
        cy.get(".field-validation-error > span").should('contain',defaultValues.InvalidLengthUsername);
    })

    it('RF_03 - Unico', () => {
        creationPage.verifyUser("1");
        creationPage.verifyUser("0");
        creationPage.verifyEnterpriseUser();
    })
   
})

File: Selenium/src/FMS Web Portal/Tests/Users/HumanUsers/Username.cs
using Common.Classes;
using FMSWebPortal.Helpers.Users;
using FMSWebPortal.PageObjects;
using FMSWebPortal.PageObjects.Users.HumanUsers;
using FMSWebPortal.Shared;
using FMSWebPortal.Shared.DataHelper;
using Xunit;

namespace Users.HumanUsers
{
    [Collection("Collection #3")]
    public class Username : TestsBase
    {
        private readonly CreationEditonPage _createPage;
        private readonly DataHelper _data;
        private readonly LoginPage _loginPage;

        public Username()
        {
            _createPage = new CreationEditonPage();
            _data = new DataHelper();
            _loginPage = new LoginPage();
            _loginPage.Navigate();
            _loginPage.FillAndSubmit();
        }


        [Fact(DisplayName = "Username - Caracteres permitidos")]
        public void RF_01()
        {
            _createPage.GoToCreationPage();
            _createPage.VerifyValidUsername();
            _createPage.VerifyInvalidUsername(RandomHelper.RandomString(invalidCharacters: true));
        }

        [Fact(DisplayName = "Username - Prefijos no permitidos")]
        public void RF_02()
        {
            _createPage.GoToCreationPage();
            _createPage.VerifyInvalidUsername("endpoint");
            _createPage.VerifyInvalidUsername("anyone");
        }

        [Fact(DisplayName = "Username - Unico")]
        public void RF_03()
        {
            _createPage.GoToCreationPage();
            _createPage.FillForm(UsersHelper.CreateBody());
            _createPage.VerifyFieldUnique(_data.GetUsername(), "Username");
            _createPage.VerifyFieldUnique(UsersHelper.GetRandomEnterpriseUser().Username, "Username");
        }

        [Fact(DisplayName = "Username - longitud permitida")]
        public void RF_04()
        {
            _createPage.GoToCreationPage();
            _createPage.VerifyValidUsername();
            _createPage.VerifyInvalidUsername(RandomHelper.RandomString(length: 65, username: true), invalid_length: true);
        }
    }
}
